zpxe.rexx

```diff
@@ -249,6 +249,49 @@ GetTFTP:
 return 0 /* GetTFTP */
  
  
+/* Procedure GetFTP
+   Use CMS FTP client to download files over anonymous FTP
+     server: remote FTP server hostname
+     path: remote file location
+     filename: local file name
+     transfermode: 'ascii' or 'binary'
+*/
+GetFTP:
+
+  parse arg ftpserver path filename transfermode
+
+  queue 'anonymous anonymous'
+  if transfermode <> '' then
+    queue transfermode
+  queue 'get' path filename
+  queue 'quit'
+
+  'set cmstype ht' /* suppress ftp output */
+  'ftp' ftpserver '(exit'
+  'set cmstype rt'
+
+return 0 /* GetFTP */
+
+
+/* Procedure GetURL
+   Download from a URL (or plain TFTP path)
+*/
+GetURL:
+
+  /* is it an FTP url? */
+  parse arg 'ftp://' ftpserver '/' path filename
+  if ftpserver <> '' & path <> '' then do
+    call GetFTP ftpserver path filename 'binary'
+    return 0
+  end
+
+  /* assume it's a TFTP path on the PXE server */
+  parse arg path filename
+  call GetTFTP server path filename 'octet'
```

Comment:
This is not right... not sure how I messed that up, I guess I was going back and forth a few times on whether the server hostname should be passed into `GetTFTP` or not. But right now it *doesn't* accept the server as an arg so it needs to be taken out here.

I was probably accidentally testing a different version of this patch, which is a bit of a worry...